kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/resources/form-templates-providers/js/kieserver-ui.js

```diff
@@ -324,38 +324,72 @@ function validate(start) {
 
 var fileData = new Map();
 
-
 function encodeImageFileAsURL(input) {
 
 	var filesSelected = input.files;
 	if (filesSelected.length > 0) {
 		var fileToLoad = filesSelected[0];
-		
+
 		var fileInfo = {
-				'name' : fileToLoad.name,
-				'size' : fileToLoad.size,
-				'data' : null
+			'name': fileToLoad.name,
+			'size': fileToLoad.size,
+			'data': null
 		};
 		fileData.set(input.id, fileInfo);
-		
+
 		var fileReader = new FileReader();
 
-		fileReader.onload = function(fileLoadedEvent) {
-			var local = fileLoadedEvent.target.result; // <--- data: base64
-			var srcData = local.replace(/^data:.*\/.*;base64,/, "");
+		fileReader.onload = function () {
+			let binary = '';
+			let buffer = new Uint8Array(fileReader.result);
+			for (let i = 0; i < buffer.byteLength; i++) {
+				binary += btoa(String.fromCharCode(buffer[i]));
+			}
+			var srcData = binary.replace(/^data:.*\/.*;base64,/, "");
 
 			fileData.get(input.id).data = srcData;
-			
-			console.log("Converted Base64 version is " + srcData);
 		}
-		fileReader.readAsDataURL(fileToLoad);
+		documentChunk(fileReader, fileToLoad)
 	} else {
 		alert("Please select a file");
 	}
 }
 
-function getDocumentData(inputId) {
+function documentChunk(fileReader, fileToLoad) {
+
+	let CHUNK_SIZE = 10 * 1024;
+	let startTime, endTime;
+	let reverse = false;
+
+	let buffer = new Uint8Array(fileReader.result);
```

Comment:
I don't understand the order of execution here.
So my understanding is that `fileReader.result` is available after a call of one of `fileReader.read*` methods.
I can see `fileReader.readAsArrayBuffer(slice)` being called through `seek()` on L381. So I am wondering what fileReader.result is on this line - where is the respective fileReader.read* called?

Reply:
fileReader.readAsArrayBuffer(slice) will trigger onload.